homeassistant/components/group/sensor.py

```diff
@@ -0,0 +1,405 @@
+SENSOR_TYPES = {
+    ATTR_MIN_VALUE: "min",
+    ATTR_MAX_VALUE: "max",
+    ATTR_MEAN: "mean",
+    ATTR_MEDIAN: "median",
+    ATTR_LAST: "last",
+    ATTR_RANGE: "range",
+    ATTR_SUM: "sum",
+}
+SENSOR_TYPE_TO_ATTR = {v: k for k, v in SENSOR_TYPES.items()}
+
+# No limit on parallel updates to enable a group calling another group
+PARALLEL_UPDATES = 0
+
+PLATFORM_SCHEMA = PARENT_PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_ENTITIES): cv.entities_domain(
+            [DOMAIN, NUMBER_DOMAIN, INPUT_NUMBER_DOMAIN]
+        ),
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_UNIQUE_ID): cv.string,
+        vol.Optional(CONF_ALL, default=False): cv.boolean,
+        vol.Optional(CONF_TYPE, default=SENSOR_TYPES[ATTR_MAX_VALUE]): vol.All(
+            cv.string, vol.In(SENSOR_TYPES.values())
+        ),
+        vol.Optional(CONF_ROUND_DIGITS, default=2): vol.Coerce(int),
+        vol.Optional(CONF_UNIT_OF_MEASUREMENT): str,
+        vol.Optional(CONF_DEVICE_CLASS): DEVICE_CLASSES_SCHEMA,
+        vol.Optional(CONF_STATE_CLASS): STATE_CLASSES_SCHEMA,
+    }
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up the Switch Group platform."""
+    async_add_entities(
+        [
+            SensorGroup(
+                config.get(CONF_UNIQUE_ID),
+                config[CONF_NAME],
+                config[CONF_ENTITIES],
+                config[CONF_ALL],
+                config[CONF_TYPE],
+                config[CONF_ROUND_DIGITS],
+                config.get(CONF_UNIT_OF_MEASUREMENT),
+                config.get(CONF_STATE_CLASS),
+                config.get(CONF_DEVICE_CLASS),
+            )
+        ]
+    )
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Initialize Switch Group config entry."""
+    registry = er.async_get(hass)
+    entities = er.async_validate_entity_ids(
+        registry, config_entry.options[CONF_ENTITIES]
+    )
+    async_add_entities(
+        [
+            SensorGroup(
+                config_entry.entry_id,
+                config_entry.title,
+                entities,
+                config_entry.options[CONF_ALL],
+                config_entry.options[CONF_TYPE],
+                int(config_entry.options[CONF_ROUND_DIGITS]),
+                None,
+                None,
+                None,
+            )
+        ]
+    )
+
+
+def calc_min(sensor_values: list[tuple[str, Any]]) -> tuple[str | None, float | None]:
+    """Calculate min value."""
+    val: float | None = None
+    entity_id: str | None = None
+    for sensor_id, sensor_value in sensor_values:
+        if sensor_value not in [STATE_UNKNOWN, STATE_UNAVAILABLE] and (
```

Comment:
Is filtering of unknown / unavailable really needed here, hasn't that already been done?